src/pudl/extract/ferc1.py

```diff
@@ -826,58 +832,71 @@ def extract_xbrl(
             f"pandas DataFrame from XBRL table."
         )
 
-        # Attempt to extract both duration and instant tables
-        xbrl_table = TABLE_NAME_MAP_FERC1[pudl_table]["xbrl"]
-        ferc1_raw_dfs[pudl_table] = {}
-        for period_type in ["duration", "instant"]:
-            ferc1_raw_dfs[pudl_table][period_type] = extract_xbrl_generic(
-                ferc1_engine=sa.create_engine(pudl_settings["ferc1_xbrl_db"]),
+        xbrl_table_or_tables = TABLE_NAME_MAP_FERC1[pudl_table]["xbrl"]
+        if not isinstance(xbrl_table_or_tables, list):
+            xbrl_tables = [xbrl_table_or_tables]
+        else:
+            xbrl_tables = xbrl_table_or_tables
+
+        ferc1_xbrl_engine = sa.create_engine(pudl_settings["ferc1_xbrl_db"])
+
+        ferc1_raw_dfs[pudl_table] = {
+            period_type: extract_xbrl_generic(
+                table_names=xbrl_tables,
+                period=period_type,
+                ferc1_engine=ferc1_xbrl_engine,
                 ferc1_settings=ferc1_settings,
-                table_name=f"{xbrl_table}_{period_type}",
             )
+            for period_type in ["duration", "instant"]
+        }
 
     return ferc1_raw_dfs
 
 
-def extract_xbrl_generic(
+def extract_xbrl_single(
+    table_name: str,
+    period: str,
     ferc1_engine: sa.engine.Engine,
     ferc1_settings: Ferc1Settings,
-    table_name: str,
 ) -> pd.DataFrame:
     """Extract a single FERC Form 1 XBRL table by name.
 
     Args:
-        ferc1_engine: An SQL Alchemy connection engine for the FERC Form 1 database.
-        ferc1_settings: Object containing validated settings relevant to FERC Form 1.
         table_name: Name of the XBRL table to extract, as it appears in the original
             XBRL derived SQLite database.
+        period: Either duration or instant, specific to xbrl data.
+        ferc1_engine: An SQL Alchemy connection engine for the FERC Form 1 database.
+        ferc1_settings: Object containing validated settings relevant to FERC Form 1.
     """
+    # Create full table name with _instant or _duration suffix
+    table_name_full = f"{table_name}_{period}"
+
     # Get XBRL DB metadata
     ferc1_meta = get_ferc1_meta(ferc1_engine)
 
     # Not every table contains both instant and duration
     # Return empty dataframe if table doesn't exist
-    if table_name not in ferc1_meta.tables:
+    if table_name_full not in ferc1_meta.tables:
         return pd.DataFrame()
 
     # Identification table used to get the filing year
     id_table = "identification_001_duration"
 
     return pd.read_sql(
         f"""
-        SELECT {table_name}.*, {id_table}.report_year FROM {table_name}
-        JOIN {id_table} ON {id_table}.filing_name = {table_name}.filing_name
+        SELECT {table_name_full}.*, {id_table}.report_year FROM {table_name_full}
```

Comment:
Out of scope and non-blocking: did you consider turning this into a SQLAlchemy select instead of a straight text select?

Reply:
I didn't write this original SQL, so I'm less familiar with this